src/optimizer.cpp

```diff
@@ -123,10 +117,41 @@ geometry_msgs::msg::TwistStamped Optimizer::evalControl(
     shiftControlSequence();
   }
 
-  if (stop_flag) {
+  fallback(critics_data_.fail_flag);
+
+  return control;
+}
+
+void Optimizer::fallback(bool fail)
```

Comment:
I don't think this is correct. If we fail, we should be doing `reset()`, right?

Then if we fail more than `N` times, then we reset and we also throw an exception. This will only reset & throw exception after `N`. I thought we wanted to reset after every failure and let it try from scratch between them?

Reply:
Yeah sure. I was a little bit inattentive here

Reply:
fixed